dataflowengine/src/main/scala/io/shiftleft/dataflowengine/language/TrackingPoint.scala

```diff
@@ -84,26 +83,22 @@ class TrackingPoint(val wrapped: NodeSteps[nodes.TrackingPoint]) extends AnyVal
       case _                                      => None
     }
 
-  private def methodFast(dataFlowObject: Vertex): nodes.Method = {
-    val method =
-      dataFlowObject.label match {
-        case NodeTypes.METHOD_RETURN =>
-          ExpandTo.methodReturnToMethod(dataFlowObject)
-        case NodeTypes.METHOD_PARAMETER_IN =>
-          ExpandTo.parameterInToMethod(dataFlowObject)
-        case NodeTypes.METHOD_PARAMETER_OUT =>
-          ExpandTo.parameterInToMethod(dataFlowObject)
-        case NodeTypes.LITERAL | NodeTypes.CALL | NodeTypes.IDENTIFIER | NodeTypes.RETURN | NodeTypes.UNKNOWN =>
-          ExpandTo.expressionToMethod(dataFlowObject)
-      }
-    method.asInstanceOf[nodes.Method]
-  }
+  // TODO match on type rather than label: depends on https://github.com/ShiftLeftSecurity/codepropertygraph/issues/610
+  private def methodFast(dataFlowObject: nodes.TrackingPoint): nodes.Method =
```

Comment:
We could 
```
match{
      dataFlowObject case methodReturn: nodes.MethodReturn =>   ExpandTo.methodReturnToMethod(methodReturn)
      case methodParameterIn: nodes.MethodParameterIn =>  ExpandTo.parameterInToMethod(methodParameterIn)
      case methodParameterOut: nodes.MethodParameterOut => ExpandTo.parameterOutToMethod(methodParameterOut)
      case expression: nodes.Expression =>  ExpandTo.expressionToMethod(expression)
      case _ => throw new NotImplementedError() //e.g. free tracking point
    }
```

Reply:
i first did that and was then too afraid since it's a little more than a refactoring. 
but now I have someone else to blame if it goes wrong, thank you :)